cpp/examples/lct_secir.cpp

```diff
@@ -47,25 +47,25 @@ int main()
                                                                 {50},  {10, 10, 5, 3, 2}, {20},         {10}};
 
     // Assert that initial_populations has the right shape.
-    if (initial_populations.size() != (int)LctState::InfectionState::Count) {
+    if ((int)initial_populations.size() != (int)LctState::InfectionState::Count) {
         mio::log_error("The number of vectors in initial_populations does not match the number of InfectionStates.");
         return 1;
     }
-    if ((initial_populations[(int)LctState::InfectionState::Susceptible].size() !=
+    if (((int)initial_populations[(int)LctState::InfectionState::Susceptible].size() !=
          LctState::get_num_subcompartments<LctState::InfectionState::Susceptible>()) ||
```

Comment:
Here the narrowing is probably correct. You could cast to size_t if you want to assume/guarantee that `get_num_subcompartments` is non-negative.